web-console/src/views/ingestion-view/ingestion-view.tsx

```diff
@@ -278,24 +283,37 @@ ORDER BY "rank" DESC, "created_time" DESC`;
     });
 
     this.taskQueryManager = new QueryManager({
-      processQuery: async capabilities => {
-        if (capabilities.hasSql()) {
-          return await queryDruidSql({
-            query: IngestionView.TASK_SQL,
+        processQuery: async capabilities => {
+          if (this.state.preferOverlord) {
+            if (capabilities.hasOverlordAccess()) {
+              const resp = await Api.instance.get(`/druid/indexer/v1/tasks`);
+              return IngestionView.parseTasks(resp.data);
+            } else if (capabilities.hasSql()) {
+              return await queryDruidSql({
+                query: IngestionView.TASK_SQL,
+              });
```

Comment:
nit, we seen to be repeating a lot of code between the two branches of the if/else block